tardis/visualization/tools/interaction_radius_plot.py

```diff
@@ -0,0 +1,461 @@
+import tardis.visualization.tools.sdec_plot as sdec
+
+import numpy as np
+import pandas as pd
+import astropy.units as u
+
+from tardis.util.base import (
+    atomic_number2element_symbol,
+    element_symbol2atomic_number,
+    species_string_to_tuple,
+    roman_to_int,
+    int_to_roman,
+)
+
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+import matplotlib.colors as clr
+import plotly.graph_objects as go
+
+
+class InteractionRadiusPlotter:
+    """
+    Plotting interface for the interaction radius plot.
+    """
+
+    def __init__(self, data, time_explosion, velocity):
+        """
+        Initialize the plotter with required data from the simulation.
+
+        Parameters
+        ----------
+        data : dict of SDECData
+            Dictionary to store data required for interaction radius plot,
+            for both packet modes (real, virtual).
+
+        time_explosion : astropy.units.Quantity
+            Time of the explosion.
+
+        velocity : astropy.units.Quantity
+            Velocity array from the simulation.
+        """
+
+        self.data = data
+        self.time_explosion = time_explosion
+        self.velocity = velocity
+        return
+
+    @classmethod
+    def from_simulation(cls, sim):
+        """
+        Create an instance of the plotter from a TARDIS simulation object.
+
+        Parameters
+        ----------
+        sim : tardis.simulation.Simulation
+            TARDIS simulation object produced by running a simulation.
+
+        Returns
+        -------
+        Plotter
+        """
+
+        return cls(
+            dict(
+                virtual=sdec.SDECData.from_simulation(sim, "virtual"),
+                real=sdec.SDECData.from_simulation(sim, "real"),
+            ),
+            sim.plasma.time_explosion,
+            sim.simulation_state.velocity,
+        )
+
+    @classmethod
+    def from_hdf(cls, hdf_fpath):
+        """
+        Create an instance of the Plotter from a simulation HDF file.
+
+        Parameters
+        ----------
+        hdf_fpath : str
+            Valid path to the HDF file where simulation is saved.
+
+        Returns
+        -------
+        Plotter
+        """
+        hdfstore = pd.HDFStore(hdf_fpath)
+        time_explosion = hdfstore['/simulation/plasma/scalars']['time_explosion'] * u.s
+        return cls(
+            dict(
+                virtual=sdec.SDECData.from_hdf(hdf_fpath, "virtual"),
+                real=sdec.SDECData.from_hdf(hdf_fpath, "real"),
+            ),
+        )
+
+    def _parse_species_list(self, species_list):
```

Comment:
we already have similar/same method in SDEC plot. why not make that public and generalize it if needed and share methods in these two classes